### ISSUE TITLE
 runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by marrco** [projectdiscovery/nuclei#3617]
### Nuclei version:
Current nuclei version: v2.9.2 (latest)

### Current Behavior:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xea5c06]

goroutine 5382 [running]:
github.com/projectdiscovery/interactsh/pkg/client.(*Client).URL(0xc00a30bd10?)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/interactsh@v1.1.1-0.20230417162754-2cd861b12467/pkg/client/client.go:569 +0x26
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh.(*Client).URL(0xc000331a20?)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/common/interactsh/interactsh.go:209 +0x8e
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh.(*Client).NewURLWithData(0xc000331a20, {0xc00018b91a, 0x12})
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/common/interactsh/interactsh.go:251 +0x2c
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh.(*Client).ReplaceWithMarker(0xc004a71438?, {0xc00018b800, 0x180}, 0xc00a30bcc0?, {0x37e9118, 0x0, 0x0})
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/common/interactsh/interactsh.go:238 +0x117
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh.(*Client).Replace(...)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/common/interactsh/interactsh.go:232
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*requestGenerator).Make(0xc00c57cbe0, {0x2870528, 0xc0064f6ff0}, 0xc009b83660, {0xc00018b800?, 0x180?}, 0xc0064f6f90, 0xc0064f6a80)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/http/build_request.go:77 +0xcb
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults.func1({0xc00018b800, 0x180}, 0xc00bfc6bc8?, 0x10aa6dc?)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/http/request.go:357 +0x165
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults(0xc003834fc0, 0xc009b83660, 0xc0064f6a80, 0xc0064f6ab0, 0xc0064f6f30)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/http/request.go:438 +0x36a
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0xc0038ec2e0, 0xc009b835e0)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/protocols/common/executer/executer.go:81 +0x383
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0x16003ea?, 0x0?, 0xc009b83580)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/core/executors.go:128 +0x296
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.2/pkg/core/executors.go:105 +0x4f8

### Expected Behavior:

not to fail

### Steps To Reproduce:

 go/bin/nuclei -no-interactsh -u 12.34.56.78 -pt http

### Anything else:

works with older version (up to 2.9.0) but not with 2.9.1 nor 2.9.2
go version go1.20.3 linux/amd64 
(but i tried with older go version, same issue)

uname -a gives:   Linux MyComputer 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

go/bin/nuclei -no-interactsh -u 12.34.56.78 -ept http     works fine so i suppose there's an issue with some http templates.

**Comment by ehsandeep** [comment]
@marrco this is a duplicate of https://github.com/projectdiscovery/nuclei/issues/3584 and fixed in the dev branch.

You can install dev branch using 

```
go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@dev
```

**Comment by marrco** [comment]
thx, I can confirm the issue is fixed in  v2.9.3-dev (development)
